frontend/src/components/EditableLabelsDescriptionListGroup.tsx

```diff
@@ -1,223 +1,245 @@
-  const reservedKeys = [
-    ...allExistingKeys.filter((key) => !labels.includes(key)),
-    ...unsavedLabels,
-  ];
-
-  const [isAddLabelModalOpen, setIsAddLabelModalOpen] = React.useState(false);
-  const [addLabelInputValue, setAddLabelInputValue] = React.useState('');
-  const addLabelInputRef = React.useRef<HTMLInputElement>(null);
-  let addLabelValidationError: string | null = null;
-  if (reservedKeys.includes(addLabelInputValue)) {
-    addLabelValidationError = 'Label must not match an existing label or property key';
-  } else if (addLabelInputValue.length > 63) {
-    addLabelValidationError = "Label text can't exceed 63 characters";
-  }
-
-  const toggleAddLabelModal = () => {
-    setAddLabelInputValue('');
-    setIsAddLabelModalOpen(!isAddLabelModalOpen);
+    while (unsavedLabels.includes(newLabel)) {
+      newLabel = `${baseLabel} ${counter}`;
+      counter++;
+    }
+
+    setUnsavedLabels((prev) => {
+      const updated = [...prev, newLabel];
+      return updated;
+    });
   };
-  React.useEffect(() => {
-    if (isAddLabelModalOpen && addLabelInputRef.current) {
-      addLabelInputRef.current.focus();
+
+  const labelErrors = validateLabels();
+  const shouldBeRed = (label: string, index: number): boolean => {
+    const firstIndex = unsavedLabels.findIndex((l) => l === label);
+
+    if (firstIndex !== index) {
+      return true;
     }
-  }, [isAddLabelModalOpen]);
-
-  const addLabelModalSubmitDisabled = !addLabelInputValue || !!addLabelValidationError;
-  const submitAddLabelModal = (event?: React.FormEvent) => {
-    event?.preventDefault();
-    if (!addLabelModalSubmitDisabled) {
-      addUnsavedLabel(addLabelInputValue);
-      toggleAddLabelModal();
+
+    if (allExistingKeys.includes(label) && !labels.includes(label)) {
+      return true;
     }
+
+    return false;
   };
 
+  // Add a ref for the alert
+  const alertRef = React.useRef<HTMLDivElement>(null);
+
   return (
-    <>
-      <DashboardDescriptionListGroup
-        title={title}
-        isEmpty={labels.length === 0}
-        contentWhenEmpty={contentWhenEmpty}
-        isEditable={!isArchive}
-        isEditing={isEditing}
-        isSavingEdits={isSavingEdits}
-        contentWhenEditing={
+    <DashboardDescriptionListGroup
+      editButtonTestId="editable-labels-group-edit"
+      saveButtonTestId="editable-labels-group-save"
+      title={title}
+      isEmpty={labels.length === 0}
+      contentWhenEmpty={contentWhenEmpty}
+      isEditable={!isArchive}
+      isEditing={isEditing}
+      isSavingEdits={isSavingEdits}
+      contentWhenEditing={
+        <>
           <LabelGroup
-            data-testid="label-group"
+            data-testid="editable-label-group"
             isEditable={!isSavingEdits}
-            numLabels={unsavedLabels.length}
+            numLabels={10}
+            expandedText="Show Less"
+            collapsedText="Show More"
```

Comment:
Ooooone last thing - we should set `numLabels` back to `unsavedLabels.length` and remove the `expandedText` and `collapsedText` props. When editing labels they should always be expanded, the show more/less behavior is only when they are not being edited which is handled by the other LabelGroup rendered in non-edit mode. Otherwise, if they are collapsed and you click "Add label" the new label doesn't appear until you expand.